### PR TITLE
PR to self for review : implement azure platform

### PR DESCRIPTION
**Pull request by serbrech**

### CODE REVIEW

File: pkg/asset/cluster/azure/azure.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+package azure
+
+import (
+	"fmt"
+
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/azure"
+)
+
+// Metadata converts an install configuration to AWS metadata.
```
Comment (juan-lee) [inline pkg/asset/cluster/azure/azure.go]: azure

Comment (serbrech) [reply pkg/asset/cluster/azure/azure.go]: yeah, I'll do a broad AWS search in the azure implementations. leave these notes for later :)

File: pkg/asset/installconfig/azure/azure.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+	"github.com/openshift/installer/pkg/types/azure"
+	"github.com/openshift/installer/pkg/types/azure/validation"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	survey "gopkg.in/AlecAivazis/survey.v1"
+)
+
+const (
+	azurePublicCloudTenantID string = "72f988bf-86f1-41af-91ab-2d7cd011db47"
```
Comment (awesomenix) [inline pkg/asset/installconfig/azure/azure.go]: so this one is actually our tenant ID :P probably need to change this, sorry about that.

File: pkg/asset/cluster/azure/azure.go (hunk 1 above)
Comment (serbrech) [reply pkg/asset/cluster/azure/azure.go]: fixed

File: pkg/asset/installconfig/azure/azure.go (hunk 2 above)
Comment (serbrech) [reply pkg/asset/installconfig/azure/azure.go]: fixed, added tenantID to the survey questions

**Pull request merged**